### ISSUE TITLE
Google OAuth2 Authentication fails

### ISSUE THREAD

**Issue by scrod** [zulip/zulip-ios#2]
Entering the domain, clicking "Login with Google", and allowing access yields a "Uh oh, something happened that we didn't expect!" message. No combination appears to work.

**Comment by scrod** [comment]
I forgot to mention that this is probably due to the app apparently ignoring my server domain and using zulip.com instead.

**Comment by timabbott** [comment]
The update to the iOS app to support letting you specify which server to talk to is waiting in Apple's app store review queue (it's been in there for about a week).

**Comment by scrod** [comment]
Thanks. For your reference I encountered this issue with version 2.2.1, released Sept 30th with a  "What's New" of "Allow setting domain for custom Zulip servers".

**Comment by timabbott** [comment]
Hmm, OK, looks like there's a report for the server too: https://github.com/zulip/zulip/issues/102

Annoying, I bet we only tested non-zulip.com servers with password auth and Google auth isn't broken or need a bit more configuration.  Can you confirm that at least password login is working for you with iOS version 2.2.1?

**Comment by timabbott** [comment]
Now that we've merged the server-side change needed for this, this should be pretty easy to fix -- we just need to add a call to fetch the google oauth ID from the server, right?

**Comment by fire** [comment]
So the current status is the ios app cannot sign in using Google Auth right?

**Comment by timabbott** [comment]
Yep, currently it Google Auth only works with zulip.com because the Google Auth public key is hardcoded in the app.  However, I think it'd probably be a <50 line patch to make the iOS app fetch the public key from the server rather than hardcoding it...

**Comment by zstarer** [comment]
Hm, I'm running into this issue.  Haven't looked at the server logs but im getting that same error.

**Comment by scrod** [comment]
Congrats on getting everything working 👍
